### PR TITLE
#184651347 A buyer should be able to add a review and a rating to a product

### PR DESCRIPTION
**Pull request by bstwagir**
#### What does this PR do?

This adds an API endpoint that a buyer will use to add a review and a rating to a purchased product. Once the request body has been validated and approved an instance of productFeedback will be generated and saved in the database in the ProductFeedback table with related metadata such as productId and userId. And an averageRating value will be generated and added to the reviewed product table, as an average of the sum of all the reviews left on the product.

#### Description of Task to be completed?

Have the following endpoints working
- `POST/api/addProductFeedback/:id`
the :id in the route represents the productId, only a registered customer may create a review, and the rating value should be kept between 1 and 5.

#### How should this be manually tested?

- Clone the repository
- Navigate to the project
- Create a .env file
- Add the following Variables in the .env file
  - JWT_SECRET, USER_SECRET_KEY, BCRYPT_KEY, TOKEN_EXPIRATION
  - SENDGRID_API_KEY, SENDER_EMAIL, GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET
  - NODE_ENV
  - DB_USERNAME
  - DB_PASSWORD
  - DB_HOST
  - DB_NAME

- In the terminal, run `npm install`, to install the required dependencies.
- RUN `npm start`
- Use Postman to test all the endpoints
- or go to http://localhost:3000/api-docs/, to test all the endpoints

#### What are the relevant pivotal tracker/Trello stories?
 #184651347

### CODE REVIEW

**Review by kagame8732 — DISMISSED** [review]
Working well!

**Review by ruberwa — CHANGES_REQUESTED** [review]
coverage and . nyc_output think can be better to be  in git ignore

**Review by kagame8732 — DISMISSED** [review]
Working well!!

**Review by ruberwa — DISMISSED** [review]

File: src/middlewares/validateProductFeedbackMiddlware.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { Product } from '../Database/models';
```
Comment (nshutip) [inline src/middlewares/validateProductFeedbackMiddlware.js]: All try/catch hooks should be handled by the asyncWrapper from utils

**Review by nshutip — DISMISSED** [review]

**Review by kagame8732 — DISMISSED** [review]

**Review by ruberwa — DISMISSED** [review]

**Review by ruberwa — DISMISSED** [review]

**Review by kagame8732 — DISMISSED** [review]
Working well!

**Review by nshutip — DISMISSED** [review]
This PR has no functionalities now, it only has .gitignore in the files changed.
Make sure to commit all your code and push it again.

**Review by JulesNIYOMWUNGERI — APPROVED** [review]

**Review by kagame8732 — APPROVED** [review]

File: src/tests/create-product.test.js
Comment (nshutip) [inline src/tests/create-product.test.js:1]: why did you turn your tests to async functions?

**Review by nshutip — APPROVED** [review]

**Pull request merged**